### PR TITLE
BUGFIX: Fix setting sub-package key in ActionRequest

### LINKED ISSUE #1819: Routing breaks with sub package key being empty string
### Description

When trying to log in to Neos with wrong credentials, an exception is thrown.

### Steps to Reproduce

1. Checkout a Neos dev dist as master
2. Submit wrong credentials in login form

#### Expected behavior

A nice error message about the credentials being wrong.

#### Actual behavior

`Could not resolve a route and its corresponding URI for the given parameters.`

### Affected Versions

Flow: master

### PR DESCRIPTION
**Pull request by kdambekalns**
The ActionRequest needs to set the sub-package key to null when an
empty string is input. This used to be the case earlier and the
handling is effectively just reverted to that behaviour by this.

Fixes #1819

### CODE REVIEW

File: Neos.Flow/Classes/Mvc/Routing/UriBuilder.php [hunk 1]
```diff
@@ -286,28 +286,25 @@ public function reset()
...
+        if (empty($packageKey) && empty($subPackageKey)) {
             $subPackageKey = $this->request->getControllerSubpackageKey();
         }
-        if ($packageKey === null) {
+        if (empty($packageKey)) {
             $packageKey = $this->request->getControllerPackageKey();
         }
+
+        $controllerArguments['@action'] = strtolower($actionName);
+        $controllerArguments['@controller'] = strtolower(empty($controllerName) ? $this->request->getControllerName() : $controllerName);
```
Comment (albe) [inline Neos.Flow/Classes/Mvc/Routing/UriBuilder.php]: Maybe replace that with an `if (empty(...))` condition and set controllerName to be in line with the rest of the code

**Comment by gerhard-boden** [comment]
Works IRL but tests fail because of wrong type given here https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Tests/Unit/Mvc/Routing/UriBuilderTest.php#L152 and here https://github.com/neos/flow-development-collection/blob/master/Neos.Flow/Tests/Unit/Mvc/Routing/UriBuilderTest.php#L224

**Comment by albe** [comment]
So, allow nullable actionName to stay 100% b/c or break compatiblity by being strict. Guess the former is the "right" way to go. We'll surely (hopefully) revisit things with 7.0

**Comment by kdambekalns** [comment]
Let's break this down:

- `$actionName` must not be empty anyway, so no nullable needed, and it does not have a default
- `$controllerArguments` must be an array if given, null is not a valid value (it is used as an array in the method!)
- `$controllerName` must be a non-empty string if given, null is only allowed to make it optional
- `$packageKey` must be a non-empty string if given, null is only allowed to make it optional
- `$subPackageKey` must be a non-empty string if given, null is only allowed to make it optional

Summary: the tests actually do it wrong, passing an array there has never been right, our tests need to be fixed IMHO.

**Comment by kdambekalns** [comment]
Related to this: https://github.com/neos/neos-development-collection/pull/2733

**Pull request merged**